src/Controller/Action/ExportSelectedProductsToCsvAction.php

```diff
@@ -0,0 +1,112 @@
+use Symfony\Component\Messenger\MessageBusInterface;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+
+final class ExportSelectedProductsToCsvAction
+{
+    use HandleTrait;
+
+    private WishlistContextInterface $wishlistContext;
+
+    private CartContextInterface $cartContext;
+
+    private FormFactoryInterface $formFactory;
+
+    private FlashBagInterface $flashBag;
+
+    private WishlistCommandProcessorInterface $wishlistCommandProcessor;
+
+    private UrlGeneratorInterface $urlGenerator;
+
+    public function __construct(
+        WishlistContextInterface $wishlistContext,
+        CartContextInterface $cartContext,
+        FormFactoryInterface $formFactory,
+        FlashBagInterface $flashBag,
+        MessageBusInterface $messageBus,
+        WishlistCommandProcessorInterface $wishlistCommandProcessor,
+        UrlGeneratorInterface $urlGenerator
+    ) {
+        $this->wishlistContext = $wishlistContext;
+        $this->cartContext = $cartContext;
+        $this->formFactory = $formFactory;
+        $this->flashBag = $flashBag;
+        $this->messageBus = $messageBus;
+        $this->wishlistCommandProcessor = $wishlistCommandProcessor;
+        $this->urlGenerator = $urlGenerator;
+    }
+
+    public function __invoke(Request $request): Response
+    {
+        $form = $this->createForm($request);
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var SplFileObject $file */
+            $file = $this->handleCommand($form);
+
+            if (null === $file) {
+                return new RedirectResponse($this->urlGenerator->generate('bitbag_sylius_wishlist_plugin_shop_wishlist_list_products'));
+            }
+
+            return $this->returnCsvFile($file);
+        }
+
+        foreach ($form->getErrors() as $error) {
+            $this->flashBag->add('error', $error->getMessage());
+        }
+
+        return new RedirectResponse($this->urlGenerator->generate('bitbag_sylius_wishlist_plugin_shop_wishlist_list_products'));
+    }
+
+    private function createForm(Request $request): FormInterface
+    {
+        $wishlist = $this->wishlistContext->getWishlist($request);
+        $cart = $this->cartContext->getCart();
+
+        $commandsArray = $this->wishlistCommandProcessor->createAddCommandCollectionFromWishlistProducts($wishlist->getWishlistProducts());
+
+        return $this->formFactory->create(WishlistCollectionType::class, ['items' => $commandsArray], [
+            'cart' => $cart,
+        ]);
+    }
+
+    private function handleCommand(FormInterface $form): ?\SplFileObject
+    {
+        $file = new \SplFileObject('php://temp', 'w');
+        $command = new ExportWishlistToCsv($form->get('items')->getData(), $file);
+
+        return $this->handle($command);
+    }
+
+    private function returnCsvFile(\SplFileObject $file): Response
+    {
+        $file->rewind();
+
+        $response = new Response($file->fread(5000));
```

Comment:
Why not `BinaryResponse`?

Reply:
php://temp is not reusable. After writing data to it, BinaryFileResponse can't find this file